internal/cephfs/volume.go

```diff
@@ -94,36 +83,38 @@ func getVolumeRootPathCeph(ctx context.Context, volOptions *volumeOptions, cr *u
 	return strings.TrimSuffix(stdout, "\n"), nil
 }
 
-func getSubVolumeInfo(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID) (Subvolume, error) {
-	info := Subvolume{}
-	err := execCommandJSON(
-		ctx,
-		&info,
-		"ceph",
-		"fs",
-		"subvolume",
-		"info",
-		volOptions.FsName,
-		string(volID),
-		"--group_name",
-		volOptions.SubvolumeGroup,
-		"-m", volOptions.Monitors,
-		"-c", util.CephConfigPath,
-		"-n", cephEntityClientPrefix+cr.ID,
-		"--keyfile="+cr.KeyFile)
+func (vo *volumeOptions) getSubVolumeInfo(ctx context.Context, volID volumeID) (*Subvolume, error) {
+	fsa, err := vo.conn.GetFSAdmin()
+	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can not fetch metadata pool for %s:", vo.FsName, err)
+		return nil, err
+	}
+
+	info, err := fsa.SubVolumeInfo(vo.FsName, vo.SubvolumeGroup, string(volID))
 	if err != nil {
 		util.ErrorLog(ctx, "failed to get subvolume info for the vol %s: %s", string(volID), err)
 		if strings.HasPrefix(err.Error(), volumeNotFound) {
-			return info, ErrVolumeNotFound
+			return nil, ErrVolumeNotFound
 		}
 		// Incase the error is other than invalid command return error to the caller.
 		if !strings.Contains(err.Error(), invalidCommand) {
-			return info, ErrInvalidCommand
+			return nil, ErrInvalidCommand
 		}
 
-		return info, err
+		return nil, err
 	}
-	return info, nil
+
+	subvol := Subvolume{
+		// FIXME: info.BytesQuota could be of type fsAdmin.Infinite
+		BytesQuota: int64(info.BytesQuota.(fsAdmin.ByteCount)),
```

Comment:
If ceph csi always expects a quota to be set on a subvolume it created you could just do something like:
```
bq, ok := info.BytesQuota.(fsAdmin.ByteCount)
if !ok {
    return nil, fmt.Errorf("some error text")
}
``` 
above.

Reply:
Oh, yes, I was lazy when I wrote this, and forgot to address the `FIXME` notes.